### ISSUE TITLE
[optim] Extend gradient descent with stochastic gradient descent

### ISSUE THREAD

**Issue by istmarc** [istmarc/tenseur#20]
Extend the current implementation of gradient descent algorithm in https://github.com/istmarc/tenseur/blob/main/ten/optim.hxx by a stochastic version that must include nesterov, dampening, etc similar to the one used by pytorch at https://docs.pytorch.org/docs/stable/generated/torch.optim.SGD.html.
